src/app/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { Dispatch, SetStateAction } from 'react';
+import * as S from 'app/components/Modal/Modal.style';
+
+const Modal = ({
+    children,
+    visible,
+    setVisible,
+}: {
+    children: React.ReactNode;
+    visible: boolean;
+    setVisible: Dispatch<SetStateAction<boolean>>;
```

Comment:
it is better to make the definition of types in a separate interface.
For example Modal: FC<ModalProps or IModal> = () =>